syntax="proto3";

package groom;


import "google/protobuf/timestamp.proto";

service Groom {
    rpc RoomRegistration(RoomRegistrationRequest) returns (RoomRegistrationResponse);
    rpc SendNewsFlash(stream NewsFlash) returns (NewsStreamStatus);

}

message RoomRegistrationRequest {
    string room_name=1;
}

message RoomRegistrationResponse {
    int32 room_id=1;
}

message NewsFlash {
    optional google.protobuf.Timestamp news_time=1;
    string news_item=2;
}

message NewsStreamStatus {
    bool success=1;
}